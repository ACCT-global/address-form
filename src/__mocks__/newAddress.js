export default {
  addressId: '1',
  addressType: 'residential',
  city: null,
  complement: null,
  country: null,
  geoCoordinates: null,
  neighborhood: null,
  number: null,
  postalCode: null,
  receiverName: null,
  reference: null,
  state: null,
  street: null,
}
