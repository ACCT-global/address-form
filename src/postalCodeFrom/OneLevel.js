import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AddressShape from '../propTypes/AddressShape'
import SelectPostalCode from './SelectPostalCode'

class OneLevel extends Component {
  render() {
    const { address, rules, onChangeAddress } = this.props

    return (
      <div>
        <SelectPostalCode
          address={address}
          rules={rules}
          onChangeAddress={onChangeAddress}
        />
      </div>
    )
  }
}

OneLevel.propTypes = {
  address: PropTypes.shape(AddressShape),
  rules: PropTypes.object.isRequired,
  onChangeAddress: PropTypes.func.isRequired,
}

export default OneLevel
