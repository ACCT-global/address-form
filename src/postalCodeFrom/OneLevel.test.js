import React from 'react'
import renderer from 'react-test-renderer'
import OneLevel from './OneLevel'
import { shallow } from 'enzyme'
import ECU from '../country/ECU'
import newAddress from '../__mocks__/newAddress'

describe('OneLevel', () => {
  it('show state options', () => {
    const address = {
      ...newAddress,
      country: 'ECU',
    }

    const tree = renderer
      .create(
        <OneLevel address={address} rules={ECU} onChangeAddress={jest.fn()} />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should change the postal code and the state', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <OneLevel
        address={{
          ...newAddress,
          country: 'ECU',
        }}
        rules={ECU}
        onChangeAddress={handleChange}
      />
    )

    const event = { target: { value: 'Azuay___0000' } }
    wrapper.find('select').simulate('change', event)

    expect(handleChange).toHaveBeenCalledWith({
      ...newAddress,
      country: 'ECU',
      postalCode: '0000',
      state: 'Azuay',
    })
  })
})
