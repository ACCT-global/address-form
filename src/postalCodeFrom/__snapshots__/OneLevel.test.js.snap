// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneLevel show state options 1`] = `
<div>
  <label>
    province
    <select
      name="state"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      />
      <option
        value="Azuay___0000"
      >
        Azuay
      </option>
      <option
        value="Bolivar___0001"
      >
        Bolivar
      </option>
      <option
        value="Canar___0002"
      >
        Canar
      </option>
      <option
        value="Carchi___0003"
      >
        Carchi
      </option>
      <option
        value="Chimborazo___0004"
      >
        Chimborazo
      </option>
      <option
        value="Cotopaxi___0005"
      >
        Cotopaxi
      </option>
      <option
        value="El Oro___0006"
      >
        El Oro
      </option>
      <option
        value="Esmeraldas___0007"
      >
        Esmeraldas
      </option>
      <option
        value="Galapagos___0008"
      >
        Galapagos
      </option>
      <option
        value="Guayas___0009"
      >
        Guayas
      </option>
      <option
        value="Imbabura___0010"
      >
        Imbabura
      </option>
      <option
        value="Loja___0011"
      >
        Loja
      </option>
      <option
        value="Los Rios___0012"
      >
        Los Rios
      </option>
      <option
        value="Manabi___0013"
      >
        Manabi
      </option>
      <option
        value="Morona Santiago___0014"
      >
        Morona Santiago
      </option>
      <option
        value="Napo___0015"
      >
        Napo
      </option>
      <option
        value="Orellana___0016"
      >
        Orellana
      </option>
      <option
        value="Pastaza___0017"
      >
        Pastaza
      </option>
      <option
        value="Pichincha___0018"
      >
        Pichincha
      </option>
      <option
        value="Santa Elena___0019"
      >
        Santa Elena
      </option>
      <option
        value="Santo Domingo De Los Tsachilas___0020"
      >
        Santo Domingo De Los Tsachilas
      </option>
      <option
        value="Sucumbios___0021"
      >
        Sucumbios
      </option>
      <option
        value="Tungurahua___0022"
      >
        Tungurahua
      </option>
      <option
        value="Zamora Chinchipe___0023"
      >
        Zamora Chinchipe
      </option>
    </select>
  </label>
</div>
`;
